### ISSUE TITLE
Use onLoad hook to initialise JVM and register RJDBC.jar

### ISSUE THREAD

**Issue by imanuelcostigan** [s-u/RJDBC#18]
A call to `rJava::.jpackage()` initialises the JVM for an R package and registers `*.jar` files. Currently, `RJDBC.jar` is registered when `RJDBC::JDBC()` is called. 

RSQLServer is using the jTDS driver but extending RJDBC's classes and methods (where necessary). In particular, I do not call `RJDBC::JDBC`. That means that RJDBC::fetch()` fails with:

```
Error in .jnew("info/urbanek/Rpackage/RJDBC/JDBCResultPull", .jcast(res@jr,  : 
  java.lang.ClassNotFoundException
```

I looks like this is because I don't call `RJDBC::JDBC()` which has the side-effect of adding `JDBCResultPull` to the class path. 

Would it be possible to add the following code to make this side-effect available to downstream dependent packages such as RSQLServer? This could be done by for e.g. including the following look hook:

```R
# Following instructions:
# http://cran.r-project.org/web/packages/helloJavaWorld/vignettes/helloJavaWorld.pdf
.onLoad <- function (libname, pkgname) {
  rJava::.jpackage(pkgname, lib.loc = libname)
}
```

**Comment by imanuelcostigan** [comment]
Of course I do call `RJDBC::JDBC()`. So back to drawing board.

**Comment by imanuelcostigan** [comment]
Issue arises because I installed the binary `RJDBC` package. As a result, in `RJDBC::JDBC()`, 
`system.file("java", "RJDBC.jar", package="RJDBC")` returns an empty string. If load hook is included in the package, this problem goes away. Can I submit a pull request with this change?

**Comment by s-u** [comment]
That just means that your installation of RJDBC is broken. Where did you get that binary from? Did you run `mkdist` to create the package tar ball?

**Comment by imanuelcostigan** [comment]
I've reinstalled binary from CRAN and it works fine. I'm sorry @s-u, I'm not sure why I was getting this issue. I would have installed the package from CRAN. Possibly due to installing R 3.2.0 and using a library that isn't R version specific? I don't know. I did try to make create the package tar ball using mkdist to see whether issue was with way the JAR was loaded, but that was after I got the error.
